python/scripts/cirkopt.py

```diff
@@ -213,68 +213,63 @@ def explore(self):
 
     # pylint: disable=no-self-use
     def search(self):
-        parser = argparse.ArgumentParser(
+        parser = configargparse.ArgumentParser(
             description="Use genetic search to find optimal design",
+            default_config_files=[os.path.join(PYTHON_DIRECTORY, "conf/search-default.ini")],
+            args_for_setting_config_path=["-c", "--config-file"],
+            config_arg_help_message="Path to config file",
+            args_for_writing_out_config_file=["--write-out-config-file"],
+            write_out_config_file_arg_help_message="Write out configuration to specified file",
```

Comment:
For my own understand, what do these mean?

Reply:
Yes you can still use command line args as usual, before I added the config file stuff I played around and it seems to work as a drop in replacement. 

default_config_files is where it will look for a config file by default. 
args_for_setting_config_path adds the option to override some of the default configs in a second config file user specified
config_arg_help_message is the help message for that new argument
args_for_writing_out_config_file is an argument you can specify that will write out your current config to a config file you specify. I used this to create the config files (it seems to default to .ini format)
write_out_config_file_arg_help_message is the help message for that argument

The config files can be in a number of different formats, including ini (which I used because that is the format it dumped), you can also use yaml, the original --argument used by arg parser, or a mix of these and a couple others.